projects/plugins/jetpack/3rd-party/wpcom-reader.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * This provides minor tweaks to improve the experience for Jetpack feed in the WordPress.com Reader.
+ *
+ * This does not make sites available in the Reader—that depends on the public access to /feed/ as a method on the WP.com side
+ * to check if a site is public. It also does not add any content to the feed. Any content that should not be displayed in the Reader
+ * or other RSS readers should be filtered out elsewhere.
+ *
+ * These hooks were originally part of the now-deprecated Enhanced Distribution.
+ *
+ * @since $$next-version$$
+ * @package Automattic/jetpack
+ */
+
+add_action( 'rss_head', 'jetpack_wpcomreader_feed_id' );
+add_action( 'rss_item', 'jetpack_wpcomreader_post_id' );
+add_action( 'rss1_head', 'jetpack_wpcomreader_feed_id' );
+add_action( 'rss1_item', 'jetpack_wpcomreader_post_id' );
+
+/**
+ * Output feed identifier based on blog ID.
+ */
+function jetpack_wpcomreader_feed_id() {
+	$id = (int) Jetpack_Options::get_option( 'id' );
+	if ( $id > -1 ) {
```

Comment:
Is this meant to check if the site is connected to WordPress.com, maybe? If so, could we update it to a more generic check like

```php
if (
	( new Host() )->is_wpcom_simple()
	|| (
		( new Connection_Manager( 'jetpack' ) )->has_connected_owner()
		&& ! ( new Status() )->is_offline_mode()
	)
) {`

And we'd then rely on `\Automattic\Jetpack\Connection\Manager::get_site_id();` to fetch the site id?

Reply:
Good point. I didn't take this as a chance to improve the code.